package/upgrade/lib.sh

```diff
@@ -184,3 +184,32 @@ detect_upgrade()
 
   UPGRADE_PREVIOUS_VERSION=$(echo "$upgrade_obj" | yq e .status.previousVersion -)
 }
+
+upgrade_addon()
+{
+  local name=$1
+  local namespace=$2
+  local version=$3
+
+  version=$(cat ./addons/${name}.yaml | yq e .spec.version)
+
+  cat > addon-patch.yaml <<EOF
+spec:
+  version: $version
+EOF
+
+  item_count=$(kubectl get addons.harvesterhci $name -n $namespace -o  jsonpath='{..name}')
+  if [ -z "$item_count" ]; then
+    install_chart $name $namespace
```

Comment:
do not find definition of `install_chart`, is it `install_addon` ?

Reply:
good spot.. fixed now.